ext/mbstring/tests/mb_internal_encoding_ini_invalid_encoding.phpt

```diff
@@ -0,0 +1,27 @@
+--TEST--
+Test INI mbstring.internal_encoding basic - invalid encoding specified in INI
+--SKIPIF--
+<?php
+extension_loaded('mbstring') or die('skip');
+function_exists('mb_stripos') or die("skip mb_stripos() is not available in this build");
+?>
+--INI--
+mbstring.internal_encoding=BAD
+--FILE--
+<?php
+
+echo "*** Testing INI mbstring.internal_encoding: invalid encoding specified in INI ***\n";
+
+echo mb_internal_encoding()."\n";
+echo ini_get('mbstring.internal_encoding')."\n";
+mb_internal_encoding('UTF-8');
+echo mb_internal_encoding()."\n";
+echo ini_get('mbstring.internal_encoding')."\n";
+
+?>
+--EXPECT--
+*** Testing INI mbstring.internal_encoding: invalid encoding specified in INI ***
+UTF-8
+BAD
+UTF-8
+BAD
```

Comment:
@nikic not sure I understood your comment about mb_internal_encoding with an INI setting so I added this tests which well gives something kinda surprising :|

Reply:
So, the exception just ends up being ignored?

Reply:
Maybe? looking at the ``mb_internal_encoding_ini_basic2.phpt`` test I'm not even sure that using ``mb_internal_encoding()`` actually changes the value of the INI setting.

Reply:
Yeah, `mbstring.internal_encoding` and `mb_internal_encoding()` use separate state :) There's `default_charset`, `internal_encoding`, `mbstring.internal_encoding` and `mb_internal_encoding()` that form a hierarchy, but are all separate. Go figure ;)

Looking at the code it looks like we just silently ignore incorrect encodings and instead assign UTF-8. Ugh.

Reply:
Funzies, maybe we should document this somewhere? Should I add a comment somewhere in mbstring or add a new file in the docs/ folder?